omdb-client/impl/src/main/java/com/garret/movies/omdb/impl/client/OmdbClientDTOImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.garret.movies.omdb.impl.client;
+
+import com.garret.movies.omdb.api.OmdbClient;
+import com.garret.movies.omdb.dto.MoviesResponse;
+import com.garret.movies.omdb.dto.OmdbMovie;
+import com.garret.movies.omdb.dto.ShortMovie;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class OmdbClientDTOImpl implements OmdbClient {
+
+    @Value("${api_url}")
+    private String url;
+    @Value("${api_key}")
+    private String apiKey;
+    private final RestTemplate restTemplate;
+
+    @Override
+    public List<ShortMovie> searchMovies(String title) {
+        int page = 1;
+        boolean hasNextPage = true;
+        List<ShortMovie> result = new ArrayList<>();
+        UriComponentsBuilder urlBuilder = UriComponentsBuilder.fromHttpUrl(url)
+                .queryParam("apikey", apiKey)
+                .queryParam("r", "json")
+                .queryParam("s", title)
+                .queryParam("page", page);
+        while (hasNextPage) {
```

Comment:
I don't think this is good place to do paging. Ideally batch job which you will add should do paging. So I think here you should retrurn something like OmdbResponse<List<ShortMovie>> which will also have page info

Reply:
@Garret-vin Agree with @mmarusich Better to return page. If some service need to read more pages he will call this method multiple times.